// @novu/shared/dist/cjs/dto/workflows/step-data.dto.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ControlsMetadata = exports.UiSchema = exports.UiSchemaProperty = exports.UiComponentEnum = exports.UiSchemaGroupEnum = void 0;
var UiSchemaGroupEnum;
(function (UiSchemaGroupEnum) {
    UiSchemaGroupEnum["IN_APP"] = "IN_APP";
    UiSchemaGroupEnum["EMAIL"] = "EMAIL";
    UiSchemaGroupEnum["DIGEST"] = "DIGEST";
    UiSchemaGroupEnum["DELAY"] = "DELAY";
})(UiSchemaGroupEnum || (exports.UiSchemaGroupEnum = UiSchemaGroupEnum = {}));
var UiComponentEnum;
(function (UiComponentEnum) {
    UiComponentEnum["MAILY"] = "MAILY";
    UiComponentEnum["TEXT_FULL_LINE"] = "TEXT_FULL_LINE";
    UiComponentEnum["TEXT_INLINE_LABEL"] = "TEXT_INLINE_LABEL";
    UiComponentEnum["IN_APP_BODY"] = "IN_APP_BODY";
    UiComponentEnum["IN_APP_AVATAR"] = "IN_APP_AVATAR";
    UiComponentEnum["IN_APP_SUBJECT"] = "IN_APP_PRIMARY_SUBJECT";
    UiComponentEnum["IN_APP_BUTTON_DROPDOWN"] = "IN_APP_BUTTON_DROPDOWN";
    UiComponentEnum["URL_TEXT_BOX"] = "URL_TEXT_BOX";
    UiComponentEnum["DIGEST_AMOUNT"] = "DIGEST_AMOUNT";
    UiComponentEnum["DIGEST_UNIT"] = "DIGEST_UNIT";
    UiComponentEnum["DIGEST_KEY"] = "DIGEST_KEY";
    UiComponentEnum["DIGEST_CRON"] = "DIGEST_CRON";
    UiComponentEnum["DELAY_TYPE"] = "DELAY_TYPE";
    UiComponentEnum["DELAY_AMOUNT"] = "DELAY_AMOUNT";
    UiComponentEnum["DELAY_UNIT"] = "DELAY_UNIT";
})(UiComponentEnum || (exports.UiComponentEnum = UiComponentEnum = {}));
class UiSchemaProperty {
}
exports.UiSchemaProperty = UiSchemaProperty;
class UiSchema {
}
exports.UiSchema = UiSchema;
class ControlsMetadata {
}
exports.ControlsMetadata = ControlsMetadata;
