// @novu/node/build/module/lib/events/events.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Events = void 0;
const novu_interface_1 = require("../novu.interface");
class Events extends novu_interface_1.WithHttp {
    async trigger(workflowIdentifier, data) {
        return await this.http.post(`/events/trigger`, {
            name: workflowIdentifier,
            to: data.to,
            payload: {
                ...data?.payload,
            },
            transactionId: data.transactionId,
            overrides: data.overrides || {},
            ...(data.actor && { actor: data.actor }),
            ...(data.tenant && { tenant: data.tenant }),
            ...(data.bridgeUrl && { bridgeUrl: data.bridgeUrl }),
        });
    }
    async bulkTrigger(events) {
        return await this.http.post(`/events/trigger/bulk`, {
            events,
        });
    }
    async broadcast(workflowIdentifier, data) {
        return await this.http.post(`/events/trigger/broadcast`, {
            name: workflowIdentifier,
            payload: {
                ...data?.payload,
            },
            transactionId: data.transactionId,
            overrides: data.overrides || {},
            ...(data.tenant && { tenant: data.tenant }),
        });
    }
    async cancel(transactionId) {
        return await this.http.delete(`/events/trigger/${transactionId}`);
    }
}
exports.Events = Events;
