// @novu/shared/dist/cjs/types/notification-templates/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowOriginEnum = exports.WorkflowTypeEnum = void 0;
const channel_1 = require("../channel");
var WorkflowTypeEnum;
(function (WorkflowTypeEnum) {
    WorkflowTypeEnum["REGULAR"] = "REGULAR";
    WorkflowTypeEnum["ECHO"] = "ECHO";
    WorkflowTypeEnum["BRIDGE"] = "BRIDGE";
})(WorkflowTypeEnum || (exports.WorkflowTypeEnum = WorkflowTypeEnum = {}));
var WorkflowOriginEnum;
(function (WorkflowOriginEnum) {
    WorkflowOriginEnum["NOVU_CLOUD"] = "novu-cloud";
    WorkflowOriginEnum["NOVU_CLOUD_V1"] = "novu-cloud-v1";
    WorkflowOriginEnum["EXTERNAL"] = "external";
})(WorkflowOriginEnum || (exports.WorkflowOriginEnum = WorkflowOriginEnum = {}));
__exportStar(require("./workflow-creation-source.enum"), exports);
